interpreter/exec/simd.ml

```diff
@@ -45,15 +57,35 @@ sig
 
   val i32x4_extract_lane : int -> t -> I32.t
 
-  val f32x4_min : t -> t -> t
-  val f32x4_max : t -> t -> t
-  val f32x4_abs : t -> t
-  val f32x4_extract_lane : int -> t -> F32.t
+  (* We need type t = t to ensure that all submodule types are S.t,
+   * then callers don't have to change *)
+  module F32x4 : SimdShape with type t = t and type u = F32.t
+  module F64x2 : SimdShape with type t = t and type u = F64.t
+end
 
-  val f64x2_min : t -> t -> t
-  val f64x2_max : t -> t -> t
-  val f64x2_abs : t -> t
-  val f64x2_extract_lane : int -> t -> F64.t
+(* The base type of a SIMD shape, e.g. for F32x4, the ElementType is F32 *)
+module type ElementType =
+sig
+  type t
+  val abs : t -> t
+  val min : t -> t -> t
+  val max : t -> t -> t
+end
+
+module MakeSimdShape (B : ElementType) (Rep : sig
```

Comment:
I guess I could move this into `module Make` below, then I don't need to have the `Rep` parameter.

Reply:
Makes sense. Same comment about renaming this to `MakeFloat`.

Reply:
I also changed ElementType to be Float.S.